### DISCUSSION TITLE
this.$router in vuex actions does not exist on type

### DISCUSSION THREAD

**Discussion by Mindik** [quasarframework/quasar#10554] [General - Components / Directives / etc]
Typescript can't define a property here, type extension doesn't help, how to fix it?

Of course, in fact, there is a property, if you use JS, then there will be no problems.

I will be glad to any ideas.
 
![image](https://user-images.githubusercontent.com/61435238/131730868-7718b470-2a3c-4a5c-a2d4-f56545a308cf.png)

![image](https://user-images.githubusercontent.com/61435238/131731115-ffb7f841-0f8f-4458-8d0b-58d050dc77a0.png)

**Comment by hawkeye64** [comment]
You "actions" should be a function, not a fat arrow (=>) if you want access to the store using `this`.

**Reply by Mindik** [reply]
I don't use arrow functions in actions. `This` is defined correctly in them, the `$router` property is present in it. But, the TS does not see it in the current `This` type. 
`(this as any).$router` - works, but it's a bad solution
